accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,514 @@
+| metadata.completeness.environment  | True (i.e builder is not allowed to use other environment variables than what’s listed in invocation.environment)                                                                                                  |
+| metadata.completeness.materials    | False. Builds could be pulling in extraneous scripts that are not all known ahead of time.                                                                                                                                                                                    |
+| metadata.reproducible              | False. Would only be possible if specific build tools are used and this would be too limiting for a general purpose npm publish action.                                                                                                                                                                          |
+| materials.uri                      | https://github.com/onwer/repo/refs/head/branch                                                                                                                                                               |
+| materials.digest                   | Git SHA                                                                                                                                                                                                            |
+
+The npm CLI will need to extract this information from the current build environment (e.g. GitHub Actions, CircleCI, GitLab PipeLines etc.).
+
+In GHA this information is available via environment variables as well as the OIDC id token (JWT). These variables should not be included in the build invocation environment, as they don't affect the build.
+
+#### Environment variables needed during provenance generation
+The environment variables below are needed to create the materials section, `configSource` and the build invocation id in the provenance attestation. Some variables listed below are also available as claims in the OIDC id token (JWT), reading those values from the JWT is preferable as it's authenticated.
+
+| What              | GitHub                     |
+|-------------------|----------------------------|
+| Run id            | GITHUB_CONTEXT.run_id      |
+| Run attempt       | GITHUB_CONTEXT.run_attempt |
+| Workflow name     | GITHUB_CONTEXT.workflow    |
+| Server URL        | GITHUB_CONTEXT.server_url  |
+| Repository        | GITHUB_CONTEXT.repository  |
+| Ref               | GITHUB_CONTEXT.ref         |
+| Git SHA           | GITHUB_CONTEXT.sha         |
+| OS image          | ImageOS                    |
+| OS image version  | ImageVersion               |
+| Host architecture | RUNNER_ARCH                |
```

Comment:
This seems GitHub-centric. Other platforms should be added with `???` for unknown values.